jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -96,11 +118,37 @@ public void sendMessage(String message) {
         // If the display message fragment is visible show the message there.
         if (displayMessageFragment != null) {
             displayMessageFragment.updateMessage(message);
-            return;
+        } else {
+            Log.w(MAIN_ACTIVITY_TAG,
+                    "Trying to send message, but displayMessageFragment is null.");
+        }
+
+        if (messageFile == null) {
```

Comment:
Use golden rule/path. That the file does not exist is the __exceptional case__ that we __guard__ against. Should the file for whatever exceptional reason not exist, we just return. For the rest of the method we can then just focus on the interesting part of the code, without being in the context of an `if`.

Eventhough `messageFile` isn't used in `writeMessageToFile`, the  "we just want to make sure the file is actually created before we write to it" is what we want a guard for, before doing the business in `writeMessageToFile`. If the check `messageFile == null` is a precondition for `writeMessageToFile`, we should put it in `writeMessageToFile`, or we will have to remember to do that check every time.

As an alternative, we could consider using the check to create the file if it is not there.